### PR TITLE
feat: add formatting on save using ruff and import sort with isort

### PR DESCRIPTION
**Pull request by signekb**
## Description

- This PR adds the automatic formatting on save (using ruff) as well as triggering isort to run.
- I find both very nice and helpful when writing code.

## Reviewer Focus

What do you think? 😉 
Could format.onSave=True be a hassle under other circumstances that I haven't thought about? E.g., with other file types?

### CODE REVIEW

**Review by philter87 — DISMISSED** [review]

File: .vscode/extensions.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
 		"streetsidesoftware.code-spell-checker",
 		"vivaxy.vscode-conventional-commits",
 		"charliermarsh.ruff",
-		"pshaddel.conventional-branch"
+		"pshaddel.conventional-branch", 
+		"ms-python.isort"
```
Comment (lwjohnst86) [inline .vscode/extensions.json:22]: Ruff already implements something that can replace isort (https://pypi.org/project/ruff/0.0.122/). Is there something you don't like with how it sorts?

**Comment by lwjohnst86** [comment]
Format on save is a great idea 🤩

File: .vscode/extensions.json (hunk 1 above)
Comment (signekb) [reply .vscode/extensions.json:22]: Yeah, I just couldn't get it to do the same sorting as `isort`. Maybe I'm missing something in my settings?

**Review by lwjohnst86 — DISMISSED** [review]

**Comment by lwjohnst86** [comment]
I've updated this so we trigger ruff to do isorting.

File: common/ruff.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+line-length = 79
+
+# In addition to the default formatters/linters, add these as well.
+extend-select = [
+  # Add the `line-too-long` rule to the enforced rule set. 
+  "E501", 
+  # Add rule that all functions have docstrings.
+  "D",
+  # Add isort to list.
+  "I
```
Comment (signekb) [inline common/ruff.toml:14]:
```suggestion
  "I"
```
Missing a quote here?

**Review by lwjohnst86 — APPROVED** [review]

**Comment by signekb** [comment]
Amazing. Very excited 🎉

**Pull request merged**